### DISCUSSION TITLE
Cache not found on next run

### DISCUSSION THREAD

**Discussion by Hermanverschooten** [actions/cache#522] [Q&A]
Hi,
I did a test run this morning nr 50, and noticed my compile step was again compiling. So I look at the output of the Cache step and it said 
```
1Run actions/cache@v2
2  with:
3    path: deps
4    key: Linux-23.0.2-1.11.3-deps-f3479f7b0edc4db67e7a21fe1b7e407fa92d843da286e3d412290e33fd8b630f
5  env:
6    otp-version: 23.0.2
7    elixir-version: 1.11.3
8    node-version: 13.11.0
9    MIX_ENV: test
10    cache-name: deps
11 Cache not found for input keys: Linux-23.0.2-1.11.3-deps-f3479f7b0edc4db67e7a21fe1b7e407fa92d843da286e3d412290e33fd8b630f
```
The run was successful, the cache was saved, and the dependent jobs found the cache, all is well.

I just pushed a new branch, the test started and again it will not find the cache, although the cache key has not changed.

`Cache not found for input keys: Linux-23.0.2-1.11.3-deps-f3479f7b0edc4db67e7a21fe1b7e407fa92d843da286e3d412290e33fd8b630f`

Any clues?

**Comment by dhadka** [comment]
@Hermanverschooten There are restrictions on when a cache can be used - https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache

I suspect since you mention branching that this is related to "branch scoping".  Some more info is also discussed in this issue - https://github.com/actions/cache/issues/79#issuecomment-552246833

**Reply by Hermanverschooten** [reply]
@dhadka You are not serious!  The proposed solution in #79 is fine, if you have no difference between testing and production builds. Building a cache based on the production run is useless.  I tore down my circleci setup to use GitHub actions, and this is a real show stopper.
